core/src/main/java/org/pentaho/di/connections/vfs/BaseVFSConnectionProvider.java

```diff
@@ -48,9 +48,21 @@ public abstract class BaseVFSConnectionProvider<T extends VFSConnectionDetails>
   }
 
   @Override public T prepare( T connectionDetails ) throws KettleException {
+    if ( !connectionDetails.getBARoles().contains( "Administrator" ) ) {
+      connectionDetails.getBARoles().add( "Administrator" );
+    }
```

Comment:
While there are no central constants, we could create local ones for these two system roles.